packages/apollo-codegen-core/src/compiler/index.ts

```diff
@@ -63,6 +63,7 @@ export interface Operation {
   }[];
   filePath: string;
   source: string;
+  sourceWithFragments?: string;
```

Comment:
Is there a reason for this to be a `?`. It seems to always be a string when returned from `generateOperationId()`. If we can remove the `?` here, then we can also clean up the `!` below.

Reply:
If I remember correctly, I made it optional because the `sourceWithFragments` only gets added when `--operationIdsPath` is specified.

Similar to the `operationId` property in this interface:

https://github.com/apollographql/apollo-tooling/blob/0735a1f840be41ea207fcf394b150270e5143ff1/packages/apollo-codegen-core/src/compiler/index.ts#L55-L59